wurst/util/StringUtils.wurst

```diff
@@ -157,6 +124,84 @@ public function string.reduce(string reduced) returns string
 public function string.reduce() returns string
 	return this.reduce(" ")
 
+public interface ForStringSplitCallback
+	function callback(string str)
+
+/** Custom Function which can split strings and runs callback for each string.
+	It can run even with no delimiter present (with single element).
+	Example usage:
+	"H001,H002".forEachIn(",") ->
+		doSomething()  */
+public function string.forEachIn(string delim, ForStringSplitCallback cb)
+	if this.length() > 0
+		int i = 0
+		int pos = this.indexOf(delim, i)
+
+		if pos == -1
+			cb.callback(this.substring(i, this.length()))
+
+		while pos != -1
+			cb.callback(this.substring(i, pos))
+			pos += 1
+			i = pos
+			pos = this.indexOf(delim, pos)
+			if pos == -1
+				cb.callback(this.substring(i, this.length()))
+
+/** Creates list of IDs converted from Int to their String value delimited by comma (using toRawCode).*/
+public function createDelimitedString(vararg int ids) returns string
```

Comment:
Where is the different to commaList ?

Reply:
That it does not use LinkedList inside the processing. Not really sure if its faster or anything, but the less using of classes the better I'd say.

Reply:
yeah at some point you have to many instantiated linked lists, when you canont have more then max jass array size lists at the same time but you maybe need 1 million

Reply:
> Not really sure if its faster or anything, but the less using of classes the better I'd say.

I wouldn't say that.

> yeah at some point you have to many instantiated linked lists, when you canont have more then max jass array size lists at the same time but you maybe need 1 million

That sounds more like an issue with your map architecture and not common use case imo. Additionally with the lua backend, those limits probably change.

Reply:
I would not limit it to "map architecture" problems, just that the bigger the map gets, the higher the requirements :)
Anyway, gonna remove this now, does not need to be in stdlib.

Reply:
Removed in 16e11dd